src/Microsoft.AspNet.Razor/Parser/HtmlMarkupParser.Block.cs

```diff
@@ -197,6 +197,23 @@ private void TagBlock(Stack<Tuple<HtmlSymbol, SourceLocation>> tags)
                         // End Tag
                         return EndTag(tagStart, tags, tagBlockWrapper);
                     case HtmlSymbolType.Bang:
+                        var nextSymbol = Lookahead();
+
+                        // If the next symbol is a text symbol and is not DOCTYPE then it's an ordinary HTML tag.
+                        if (nextSymbol.Type == HtmlSymbolType.Text &&
+                            !string.Equals(nextSymbol.Content, "DOCTYPE", StringComparison.OrdinalIgnoreCase))
```

Comment:
For a comment `nextSymbol.Type` is `HtmlSymbolType.DoubleHyphen`:
https://github.com/aspnet/Razor/blob/disableths.187/src/Microsoft.AspNet.Razor/Tokenizer/Symbols/HtmlSymbolType.cs#L19

Therefore it falls through to its normal logic (calling into `BangTag()`).